src/logic/calculate.js

```diff
@@ -0,0 +1,99 @@
+import operate from './operate';
+
+const calculate = ({ total, next, operation }, buttonName) => {
+  if (buttonName.match(/[0-9]/)) {
+    return {
+      total,
+      next: next ? `${next}${buttonName}` : `${buttonName}`,
+      operation,
+    };
+  }
+
+  if (buttonName === '+/-' && next) {
+    return {
+      total,
+      next: (-1 * parseFloat(next)).toString(),
+      operation,
+    };
+  }
+
+  if (buttonName === '+/-' && !next && total) {
+    return {
+      total: (-1 * parseFloat(total)).toString(),
+      next,
+      operation,
+    };
+  }
+
+  if (buttonName.match(/[-/X+]/) && next && total) {
+    return {
+      total: (operate(parseFloat(total), parseFloat(next), buttonName)).toString(),
+      next: null,
+      operation: buttonName,
+    };
+  }
+
+  if (buttonName.match(/[-/X+]/) && !next && total) {
+    return {
+      total,
+      next,
+      operation: buttonName,
+    };
+  }
+
+  if (buttonName.match(/[-/X+]/) && next && !total) {
+    return {
+      total: next,
+      next: null,
+      operation: buttonName,
+    };
+  }
```

Comment:
Make sure to use what you imported from `operate.js` which covered these cases

Reply:
Hi @fatymahmed, I am using operate function (what imported from operate.js), so your comment is not clear. An additional explanation is required. Thanks :)

Reply:
Hi @HADDADSOHAIB, you are right, you are already using the `operate` from `operate.js`, my apologies, I missed that.